odoo/tests/common.py

```diff
@@ -348,6 +348,22 @@ def __init__(self, methodName='runTest'):
         self.addTypeEqualityFunc(etree._Element, self.assertTreesEqual)
         self.addTypeEqualityFunc(html.HtmlElement, self.assertTreesEqual)
 
+    def run(self, result):
+        tests_run_count = int(os.environ.get('ODOO_TEST_FAILURE_RETRIES', 0)) + 1
+        failure = False
+        tests_run_count = 2 # TODO remove me
+        for retry in range(tests_run_count):
+            if retry:
+                _logger.runbot(f'Retrying a failed test: {self}')
+            if retry < tests_run_count-1:
+                with result.soft_fail(), quiet_logging() as quiet_log:
+                    super().run(result)
+                    failure = result.had_failure or quiet_log.had_error_log
```

Comment:
What about using an alternative object for `result` instead?  Something like
```python
local_result = SoftResult(result)       # behaves as a proxy to result
with lower_logging(25) as log:
    super().run(local_result)
    failure = local_result.has_failed or log.has_error
```
This would avoid plumbing `OdooTestResult` with a feature that you generally don't use.